lib/protobuf/dsl/typespecs.ex

```diff
@@ -80,7 +80,8 @@ defmodule Protobuf.DSL.Typespecs do
       prop.repeated? ->
         quote do: [unquote(spec)]
 
-      prop.embedded? or (prop.optional? and is_nil(prop.oneof) and syntax != :proto3) ->
+      prop.embedded? or (prop.optional? and is_nil(prop.oneof) and syntax != :proto3) or
+          (prop.proto3_optional? and syntax == :proto3) ->
```

Comment:
Can `proto3_optional?` be true when `syntax != :proto3`?
```suggestion
          prop.proto3_optional? ->
```

Reply:
Opss, you are right! `proto3_optional?` can only be true if the message uses the proto3 syntax.

✔️ Fixed, thank you!